community/portworx/templates/storage-cluster.yaml

```diff
@@ -0,0 +1,205 @@
+{{- if or (not (lookup "apps/v1" "DaemonSet" "kube-system" "portworx")) (eq true .Values.generateSTCForMigration) }}
+
+  {{- $telemetry := .Values.telemetry | default false }}
+  {{- $isCoreOS := .Values.isTargetOSCoreOS | default false }}
+  {{- $etcdEndPoints := .Values.kvdb }}
+  {{- $usefileSystemDrive := .Values.usefileSystemDrive | default false }}
+  {{- $usedrivesAndPartitions := .Values.usedrivesAndPartitions | default false }}
+  {{- $secretType := .Values.secretType | default "k8s" }}
+  {{- $deployEnvironmentIKS := .Capabilities.KubeVersion.GitVersion | regexMatch "IKS" }}
+  {{- $drives := .Values.drives | default "none" }}
+  {{- $dataInterface := .Values.dataInterface | default "none" }}
+  {{- $managementInterface := .Values.managementInterface | default "none" }}
+  {{- $envVars := .Values.envVars | default "none" }}
+  {{- $customRegistryURL := .Values.customRegistryURL | default "none" }}
+  {{- $registrySecret := .Values.registrySecret | default "none" }}
+  {{- $licenseSecret := .Values.licenseSecret | default "none" }}
+  {{- $kvdbDevice := .Values.kvdbDevice | default "none" }}
+  {{- $journalDevice := .Values.storage.journalDevice | default "none" }}
+  {{- $metadataSize := int64 .Values.storage.metadataSize | int64 }}
+  {{- $advOpts := .Values.advOpts | replace "+" "," | replace ";" " " | default "none" }}
+  {{- $icrRegistry := .Values.icrRegistry | default false }}
+  {{- $icrSecret := .Values.icrSecret | default "default/all-icr-io" }}
+  {{- $csiCloudDrive := .Values.csiCloudDrive | default false }}
+  {{- $internalKVDB := .Values.internalKVDB | default false }}
+  {{- $etcdSecret := .Values.etcd.secret | default "none" }}
+  {{- $changePortRange := .Values.changePortRange | default false }}
+  {{- $pvcControllerEnabled := .Values.enablePVCController | default false }}
+
+kind: StorageCluster
+apiVersion: core.libopenstorage.org/v1
+metadata:
+  name: "{{ required "Clustername cannot be empty" .Values.clusterName }}"
+  namespace: kube-system
+  annotations:
+    {{- if or (ne $advOpts "none") (ne $etcdSecret "none") }}
+      {{- if ne $etcdSecret "none" }}
+    portworx.io/misc-args: "-userpwd $(ETCD_USERNAME):$(ETCD_PASSWORD) -ca /etc/pwx/etcdcerts/ca.pem {{ cat $advOpts | replace "none" ""}}"
+      {{- else }}
+    portworx.io/misc-args: "{{ $advOpts }}"
+      {{- end }}
+    {{- end }}
+    {{- if eq true $pvcControllerEnabled }}
+    portworx.io/pvc-controller: "true"
+    {{- end }}
+    portworx.io/is-iks: "true"
+    portworx.io/migration-approved: "false"
```

Comment:
curious to know when migrate using helm, how does the migration work? do we just create a new stc using this template? that case how do we handle this flag?

Reply:
yes, we will create the stc here, then operator finds the stc exists and it will not create a new one. the migration thread sees the daemonset exists so it will wait for approval before starting migration